app/api/resources/user.py

```diff
@@ -15,13 +15,59 @@
 from app.api.email_utils import send_email_verification_message
 from app.api.models.user import *
 from app.api.dao.user import UserDAO
+from app.database.models.user import UserModel
 from app.api.resources.common import auth_header_parser
+from google.oauth2 import id_token
+from google.auth.transport import requests
 
 users_ns = Namespace("Users", description="Operations related to users")
 add_models_to_namespace(users_ns)
 
 DAO = UserDAO()  # User data access object
 
+def create_tokens_for_new_user_and_return(user):
+    # create tokens and expiry timestamps
+    access_token = create_access_token(identity=user.id)
+    refresh_token = create_refresh_token(identity=user.id)
+
+    from run import application
+    access_expiry = datetime.utcnow() + application.config.get(
+        "JWT_ACCESS_TOKEN_EXPIRES"
+    )
+    refresh_expiry = datetime.utcnow() + application.config.get(
+        "JWT_REFRESH_TOKEN_EXPIRES"
+    )
+
+    # return data
+    return (
+        {
+            "access_token": access_token,
+            "access_expiry": access_expiry.timestamp(),
+            "refresh_token": refresh_token,
+            "refresh_expiry": refresh_expiry.timestamp(),
+        },
+        HTTPStatus.OK,
+    )
+
+def perform_social_sign_in_and_return_response(email:str, provider: str):
+    # get existing user
+    user = DAO.get_user_by_email(email)
+
+    # if user not found, create a new user
+    if not user:
+        data = request.json
+        user = DAO.create_user_using_social_login(data, provider)
+        # If any error occured, return error
+        if not isinstance(user, UserModel):
+            return user
```

Comment:
You returning a user instance here? Is this expected? in case of error, shouldn't you format it or does the user variable have the error response in which case you should also return error type (and also can you make that clear using comments?)

Reply:
Done

Reply:
Should we unwrap the error and http code from `user` variable here to make the code more readable without comment?

Reply:
Done